docs/pages/providers/battlenet.md

```diff
@@ -0,0 +1,65 @@
+---
+title: "Battle.net"
+---
+
+# Battle.net
+
+OAuth 2.0 provider for Battle.net.
+
+Also see the [OAuth 2.0](/guides/oauth2) guide.
+
+## Initialization
+
+```ts
+import { BattleNet } from "arctic";
```

Comment:
With v3, I decided to use namespace imports for library imports.
```ts
import * as arctic from "arctic";

const battlenet = new arctic.BattleNet(clientId, clientSecret, redirectURI);
```

Can you update `generateState()` and `OAuth2RequestError` as well?